### PR TITLE
Added fn from_ansi_str for StyledObject

### PR DESCRIPTION
**Pull request by BppleMan**
this is method can parse a string with ansi escape code and return a vector of StyledObject

### CODE REVIEW

**Comment by mitsuhiko** [comment]
This looks good but I think it would be better to change this to return an Iterator instead. Also please add an example docstring showing this is to be used, maybe just use the example from #166.

**Comment by BppleMan** [comment]
Refactor parser into iterator, and provide it together with AnsiCodeIterator following ansi_parse feature, optimize unit test

**Comment by BppleMan** [comment]
Any suggestions on where to add more documentation?

**Comment by BppleMan** [comment]
@mitsuhiko can you review this pr

File: Cargo.toml [hunk 1]
```diff
@@ -21,6 +21,7 @@ ansi-parsing = []
 libc = "0.2.30"
 unicode-width = { version = "0.1", optional = true }
 lazy_static = "1.4.0"
+regex = "1.4.2"
```
Comment (mitsuhiko) [inline Cargo.toml:24]: This should be an optional dependency. Code that depends on it then should also be feature gated.

File: src/utils.rs [hunk 2]
```diff
@@ -419,6 +421,23 @@ impl Style {
     }
 }
 
+impl Style {
+    pub fn get_fg(&self) -> Option<Color> {
```
Comment (mitsuhiko) [inline src/utils.rs:425]: Why are these APIs added? Let's not add these unless there is a reason (maybe follow up PR), and if they are added, they need docs.

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

**Pull request closed without merging**